homeassistant/components/systemmonitor/sensor.py

```diff
@@ -76,17 +77,108 @@
 SIGNAL_SYSTEMMONITOR_UPDATE = "systemmonitor_update"
 
 
+def get_update(entity: SystemMonitorSensor[dataT]) -> StateType:  # noqa: C901
+    """Get the latest system information."""
+    state = None
+    type_ = entity.entity_description.key
+
+    if type_ == "disk_use_percent":
```

Comment:
This should not be a single function with checks for all entities. Each entity description should have a function that does needed parsing on the coordinator data.

Reply:
Yes. Was just parking it yesterday. Will be individual functions.